core/chains/evm/txmgr/orm.go

```diff
@@ -42,6 +42,78 @@ type orm struct {
 
 var _ ORM = (*orm)(nil)
 
+// Structs saved in database should be internal to
+// the ORM and distinct from generic params
+// so that modifying generic params does neccessitate
+// modification of underlying column types
+// (or runtime errors by mistake)
+type dbReceipt struct {
+	ID               int64
+	TxHash           common.Hash
+	BlockHash        common.Hash
+	BlockNumber      int64
+	TransactionIndex uint
+	Receipt          evmtypes.Receipt
+	CreatedAt        time.Time
+}
+
+type dbRawReceipt = evmtypes.Receipt
+
+type dbReceiptPlus struct {
+	ID           uuid.UUID        `db:"id"`
+	Receipt      evmtypes.Receipt `db:"receipt"`
+	FailOnRevert bool             `db:"FailOnRevert"`
+}
+
+func fromDBReceipts(rs []dbReceipt) (receipts []EvmReceipt) {
+	for i := 0; i < len(rs); i++ {
+		r := rs[i]
+		receipts = append(receipts, EvmReceipt{
+			ID:               r.ID,
+			TxHash:           r.TxHash,
+			BlockHash:        r.BlockHash,
+			BlockNumber:      r.BlockNumber,
+			TransactionIndex: r.TransactionIndex,
+			Receipt:          r.Receipt,
+			CreatedAt:        r.CreatedAt,
+		})
+	}
+	return
+}
```

Comment:
@prashantkumar1982 and I want to have the convention that all structs that are directly used for sqlx methods are defined in the ORM. 

So for example if we change the EvmReceipt type from
```
type EvmReceipt = txmgrtypes.Receipt[evmtypes.Receipt, common.Hash]
```
to
```
type EvmReceipt = txmgrtypes.Receipt[evmtypes.Receipt, HashInterface]
```
or
```
type EvmReceipt = txmgrtypes.Receipt[evmtypes.Receipt, int]
```

you're guarenteed to get a compile time error rather than a runtime error that tells you you're trying to save int as a byte. 

Prashant's going to replace common.Hash for an interface. Before this change he would already get a compile error but this makes the process straightforward and more legible.

Reply:
Is there any way to do this without introducing extra allocations and copying on database queries?

Reply:
In Go, structs with identical field sets can be assigned directly. IIUC that at least reduces it to one mem copy/move of the whole struct. Not sure if you can 'convert' them in-place without copying anything though :thinking:

Reply:
https://go.dev/ref/spec#Conversions

Looks like you can convert as long as everything but the struct tags are the same.